### PR TITLE
Implement nanmean function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanmean, which computes the mean of elements along a specified axis while ignoring NaN values in the input tensor.
The operation follows NumPy semantics for axis and keepdims and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/tensorflow/numpy.py [hunk 1]
```diff
@@ -2145,6 +2145,30 @@ def nanmax(x, axis=None, keepdims=False):
...
+        return x
+
+    dtype = dtypes.result_type(standardize_dtype(x.dtype), float)
+
+    nan_sum = cast(nansum(x, axis=axis, keepdims=keepdims), dtype)
+    normalizer = tf.reduce_sum(
+        cast(~tf.math.is_nan(cast(x, config.floatx())), dtype),
+        axis=axis,
+        keepdims=keepdims,
+    )
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py:2165]: Updated

File: keras/src/backend/tensorflow/numpy.py [hunk 2]
```diff
@@ -2145,6 +2145,37 @@ def nanmax(x, axis=None, keepdims=False):
...
+    if axis == () or axis == []:
+        return x
+
+    dtype = dtypes.result_type(standardize_dtype(x.dtype), float)
+
+    nan_sum = cast(nansum(x, axis=axis, keepdims=keepdims), dtype)
+
+    is_nan_mask = (
+        tf.math.is_nan(x)
+        if x.dtype.is_floating
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: I think you should just do what you did for the other ones by doing this line 2155:
```python
if not x.dtype.is_floating:
    return tf.reduce_mean(tf.cast(x, "float32"),  axis=axis, keepdims=keepdims)
```

File: keras/src/backend/tensorflow/numpy.py [hunk 3]
```diff
@@ -2145,6 +2145,37 @@ def nanmax(x, axis=None, keepdims=False):
...
+        keepdims=keepdims,
+    )
+
+    out = tf.math.divide_no_nan(nan_sum, normalizer)
+
+    return tf.where(
+        normalizer > 0,
+        out,
+        tf.fill(tf.shape(out), tf.constant(float("nan"), out.dtype)),
+    )
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: Wait, can't you just do:
```
return tf.divide(nan_sum, normalizer)
```
so that you get nan where all the numbers where nans (i.e. where normalizer is zero)?

File: keras/src/backend/tensorflow/numpy.py (hunk 2 above)
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: Thanks. I updated it.

File: keras/src/backend/tensorflow/numpy.py (hunk 3 above)
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: Thank you for pointing that out. I’ve updated it to use that directly.
